### PR TITLE
[MOV] tests: split tests files by category

### PR DESCRIPTION
**Pull request by hokolomopo**
## Description:

This commit moves the tests files to have a folder for each feature, and have all the tests for this feature in the same place, rather than the old directory structure plugin/component.

Task: : [3486271](https://www.odoo.com/web#id=3486271&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

**Comment by hokolomopo** [comment]
@LucasLefevre what do you think ? https://github.com/odoo/o-spreadsheet/pull/2660#discussion_r1297113613

**Review by LucasLefevre — COMMENTED** [review]
I like it :)

- I'd remove this extra `test_units` folder; and move everything one level up. It's ok if `test_helpers` is mixed with other folders IMO.
- I'd also rename files testing plugins by post-fixing them with `_plugin` or `_model`.
To make sure it's clear what's supposed to be in this file (no component tests). e.g. `autofill.test.ts` => `autofill_plugin.test.ts` or `autofill_model.test.ts`

**Comment by hokolomopo** [comment]
@LucasLefevre I moved everything up one folder. Kidna strange to have the tests at the same level as `setup` and `test_helper` but maybe you're right and we don't care.

**Review by LucasLefevre — COMMENTED** [review]
- postfix to be able to easilty search file names
- headers category

**Comment by VincentSchippefilt** [comment]
I like that all the tests for one specific feature are colocated. with the postfix on the file to identify which kind of tests should go where.
I don't like that fringe features now have nowhere to live b.e. the folders `misc_plugins`, `misc_components`, `components` are weird "fourre tout". 
The `bottom_bar`, `headers`, `menus` folders are all part of the "chrome" (that's the name of everything that isn't content on an app), their file could be renamed to express the kind of test they contain, and they could all be moved under a "chrome" folder

**Comment by VincentSchippefilt** [comment]
We should also write a small readme.md in the test folders to ensure that we document how we choose which test goes in which file and how files should be named, and check ourselves to ensure all our testfiles fit what we document.

File: tests/readme.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# install dependencies
+npm install
+# run the test suite
+npm run test
+```
+
+## Writing tests guidelines
+
+- Tests should be located in `tests/` folder.
+- Test files should be named `test_file_name.test.ts`
```
Comment (LucasLefevre) [inline tests/readme.md]:
```suggestion
- Test files should be suffixed with `.test.ts`. e.g `renderer_plugin.test.ts`
```

File: tests/grid/grid_drag_and_drop.test_component.test.ts
Comment (LucasLefevre) [inline tests/grid/grid_drag_and_drop.test_component.test.ts:1]: file name foireux `grid_drag_and_drop.test_component.test.ts`

File: tests/helpers/color_helpers.test.ts
Comment (LucasLefevre) [inline tests/helpers/color_helpers.test.ts:1]: can be moved to the `colors` folder

File: tests/helpers/content_editable_helpers.test.ts
Comment (LucasLefevre) [inline tests/helpers/content_editable_helpers.test.ts:1]: can be moved to composer

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Pull request closed without merging**